cep-????.md

```diff
@@ -0,0 +1,131 @@
+# CEP ???? - Virtual packages
+
+<table>
+<tr><td> Title </td><td> Virtual packages </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Jaime Rodríguez-Guerra &lt;jaime.rogue@gmail.com&gt;</td></tr>
+<tr><td> Created </td><td> Dec 17, 2024</td></tr>
+<tr><td> Updated </td><td> Dec 17, 2024</td></tr>
+<tr><td> Discussion </td><td> https://github.com/conda/ceps/pull/103 </td></tr>
+<tr><td> Implementation </td><td> https://github.com/conda/conda/tree/24.11.1/conda/plugins/virtual_packages, https://github.com/mamba-org/mamba/blob/libmamba-2.0.5/libmamba/src/core/virtual_packages.cpp, https://github.com/conda/rattler/tree/rattler-v0.28.8/crates/rattler_virtual_packages/src </td></tr>
+</table>
+
+> The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT",
+  "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as
+  described in [RFC2119][RFC2119] when, and only when, they appear in all capitals, as shown here.
+
+## Abstract
+
+This CEP standardizes which virtual packages MUST be offered by conda solvers.
+
+## Specification
+
+A virtual package is defined as a package record with three fields: name, version and build string.
+The name MUST start with double underscore. The version and build string MUST follow the same semantics as in regular package records.
+
+In general, the version or build string of a virtual package MAY be overridden by the value of `CONDA_OVERRIDE_{NAME}` environment variable, with `{NAME}` being the uppercased name of the virtual package. Many exceptions apply so please observe the details in the section below.
+
+### List of virtual packages
+
+In alphabetical order, every conda client MUST support the following virtual packages:
+
+- `__archspec`
+- `__cuda`
+- `__glibc`
+- `__linux`
+- `__osx`
+- `__unix`
+- `__win`
+
+#### `__archspec`
+
+This virtual package MUST be always present, with the version set to `1`. The build string SHOULD reflect the detected CPU microarchitecture. If it cannot be detected, the build string SHOULD be `0`.
```

Comment:
I think we'll need to standardize the acceptable values for the build string here, but that's currently tied to the `archspec` Python package implementation. Or is that list based on something already standardized? 🤔 

Also, there will be more microarchitectures in the future... will we keep updating the list? Or, alternatively, should we just defer that to whatever frontend end-users end up using ("in the end" just to use the word "end" one more time)?

Reply:
We could list some as examples and leave the rest out

That said, maybe it is a good exercise to go through updating the CEP before updating tooling, etc.?

Would be interested to hear what others think on this question. It is a good one

Reply:
Expanded this quite a bit now.